.github/actions/deploy-cloud-run/action.yml

```diff
@@ -45,11 +66,143 @@ runs:
+      id: test_live
+      with:
+        url: ${{ steps.deploy.outputs.url }}${{ inputs.healthcheck_path }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
+        customHeaders: ${{ steps.set_header.outputs.customHeaders }}
+    - name: Logging - Deploy to Cloud Run Blue
+      run: echo "Deploying ${{ inputs.image_internal || inputs.image }} to ${{ inputs.service }} blue..."
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      shell: bash
+    - name: Deploy to Cloud Run Blue
+      uses: google-github-actions/deploy-cloudrun@v2
+      id: deploy_blue
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        tag: blue-release
+        no_traffic: true
+        skip_default_labels: true
+    - name: Sleep 10 seconds before doing test on blue
+      run: |
+        echo "Sleeping 10 seconds before doing test on blue"
+        sleep 10
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      shell: bash
+    - name: Run HTTP health check on blue
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'true' && inputs.healthcheck_path }}
+      id: test_blue
+      with:
+        url: ${{ steps.deploy_blue.outputs.url }}${{ inputs.healthcheck_path }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
+        customHeaders: ${{ steps.set_header.outputs.customHeaders }}
+    - name: Logging - Check HTTP Status from test_blue
+      run: |
+        echo "HTTP Status from test_blue: ${{ steps.test_blue.outputs.status }}"
+      if: ${{ inputs.blue_green_enabled == 'true' && inputs.healthcheck_path }}
+      shell: bash
+    - name: Logging - Deploy to Cloud Run Green
+      run: echo "Deploying ${{ inputs.image_internal || inputs.image }} to ${{ inputs.service }} green..."
+      if: ${{ inputs.blue_green_enabled == 'true' && steps.test_blue.outputs.status >= 200 && steps.test_blue.outputs.status < 300 }}
+      shell: bash
+    - name: Deploy to Cloud Run Green
+      uses: google-github-actions/deploy-cloudrun@v2
+      id: deploy_green
+      if: ${{ inputs.blue_green_enabled == 'true' && steps.test_blue.outputs.status >= 200 && steps.test_blue.outputs.status < 300 }}
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        revision_traffic: 'LATEST=100'
+        skip_default_labels: 'true'
+        update_traffic_flags: '--remove-tags blue-release'
+    - name: Sleep 10 seconds before doing test on green
+      run: |
+        echo "Sleeping 10 seconds before doing test on green"
+        sleep 10
```

Comment:
We would like to consider a better method since startup time varies depending on the application.

Reply:
removed the sleep steps